File: agent/commands-generator/src/main/java/org/matilda/commands/di/ProcessorsModule.java
package org.matilda.commands.di;

import dagger.Module;
import dagger.Provides;
import org.matilda.commands.info.ServiceInfo;
import org.matilda.commands.processors.Processor;
import org.matilda.commands.processors.ProcessorFactory;

@Module
public class ProcessorsModule {
    @Provides
    Processor<ServiceInfo> serviceProcessor(ProcessorFactory factory) {
        return factory.createServiceProcessor();
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/di/ProcessorsModule.kt
package org.matilda.commands.di

import dagger.Module
import dagger.Provides
import org.matilda.commands.info.ProjectServices
import org.matilda.commands.processors.Processor
import org.matilda.commands.processors.ProcessorFactory

@Module
class ProcessorsModule {
    @Provides
    fun servicesProcessor(factory: ProcessorFactory): Processor<ProjectServices> {
        return factory.createProcessor()
    }
}
